datadog.tf

```diff
@@ -0,0 +1,133 @@
+locals {
+  datadog_operator_set = merge({
+    "site"                      = "datadoghq.eu",
+    "resources.requests.cpu"    = "10m",
+    "resources.requests.memory" = "50Mi",
+  }, var.datadog_operator_values)
+
+  datadog_agent_manifest_features = try(indent(2, join("", ["\n", var.datadog.agent_manifest_features])), indent(2, <<-YAML
+
+    features:
+      apm:
+        enabled: true
+      logCollection:
+        enabled: true
+  YAML
+  ))
+
+  datadog_agent_manifest_override = try(indent(2, join("", ["\n", var.datadog.agent_manifest_override])), indent(2, <<-YAML
+
+    override:
+      clusterAgent:
+        containers:
+          cluster-agent:
+            resources:
+              requests:
+                cpu: 30m
+                memory: 200Mi
+              limits:
+                memory: 300Mi
+      nodeAgent:
+        containers:
+          process-agent:
+            resources:
+              requests:
+                cpu: 10m
+                memory: 64Mi
+              limits:
+                memory: 128Mi
+          trace-agent:
+            resources:
+              requests:
+                cpu: 35m
+                memory: 56Mi
+              limits:
+                memory: 100Mi
+  YAML
+  ))
+}
+
+
+resource "kubernetes_secret" "datadog_keys" { # TODO: do we need this also in AWS secretsmanager?
+  count      = var.enable_datadog ? 1 : 0
+  depends_on = [helm_release.datadog]
+  metadata {
+    name      = "datadog-keys"
+    namespace = "monitoring"
+  }
+
+  data = {
+    api-key = datadog_api_key.datadog_agent[0].key
+    app-key = datadog_application_key.datadog_agent[0].key
+  }
+}
+# Datadog Operator
+resource "datadog_api_key" "datadog_agent" {
+  count = var.enable_datadog ? 1 : 0
+  name  = try(var.datadog.agent_api_key_name, local.stack_name)
+}
+
+resource "datadog_application_key" "datadog_agent" {
+  count = var.enable_datadog ? 1 : 0
+  name  = try(var.datadog.agent_app_key_name, local.stack_name)
+}
+
+resource "helm_release" "datadog" {
```

Comment:
The choice to use helm_release and not the addon module is because in the end I did not see any advantage in using the module. The addon module complicates the `set` for overriding helm values and the `create=false` still requires the `count` if there `set` or `values` depend on variables.